libbeat/common/transport/tlscommon/config.go

```diff
@@ -96,6 +98,8 @@ func LoadTLSConfig(config *Config) (*TLSConfig, error) {
 // Validate values the TLSConfig struct making sure certificate sure we have both a certificate and
 // a key.
 func (c *Config) Validate() error {
+	cfgwarn.Deprecate("8.0.0", "Treating the CommonName field on X.509 certificates as a host name when no Subject Alternative Names are present is going to be removed. Please update your certificates if needed.")
```

Comment:
Unfortunately we don't know the exact version yet, as we are subject to changes in golang. But we could implement the check ourselves.

Basically run with `Certificate` verification mode only, but install a verification step that extracts the hostname from `CommonName` and does the lookup.